### PR TITLE
[CI] Revamp VecSpec

### PR DESCRIPTION
**Pull request by jackkoenig**
* Change many tests from simulation-based to checking the CHIRRTL.
* Move tests that were hard to convert to new VecIntegrationSpec in integration-tests

Before this change, `VecSpec` took over 6 minutes on my laptop, now it takes under 10s and `VecIntegrationSpec` takes ~70s but will be able to run in parallel to the main tests in CI.

This reduced total CI test time for the 2.12.17 tests from 35m37s to 21m4s
It only increased the integration tests from 8m50s to 9m12s--the runners are dual core so there's probably some parallelism being exploited resulting in the relatively modest time increase.

### Contributor Checklist

- [NA] Did you add Scaladoc to every public function/method?
- [x] Did you add at least one test demonstrating the PR?
- [x] Did you delete any extraneous printlns/debugging code?
- [x] Did you specify the type of improvement?
- [ ] Did you add appropriate documentation in `docs/src`?
- [x] Did you state the API impact?
- [x] Did you specify the code generation impact?
- [x] Did you request a desired merge strategy?
- [NA] Did you add text to be included in the Release Notes for this change?

#### Type of Improvement

- CI performance improvement

#### API Impact

No impact

#### Backend Code Generation Impact

No impact

#### Desired Merge Strategy

- Squash

### Reviewer Checklist (only modified by reviewer)
- [ ] Did you add the appropriate labels?
- [ ] Did you mark the proper milestone (Bug fix: `3.5.x` or `3.6.x` depending on impact, API modification or big change: `5.0.0`)?
- [ ] Did you review?
- [ ] Did you check whether all relevant Contributor checkboxes have been checked?
- [ ] Did you do one of the following when ready to merge:
  - [ ] Squash: You/ the contributor `Enable auto-merge (squash)`, clean up the commit message, and label with `Please Merge`.
  - [ ] Merge: Ensure that contributor has cleaned up their commit history, then merge with `Create a merge commit`.

### CODE REVIEW

File: src/test/scala/chiselTests/Vec.scala [hunk 1]
```diff
@@ -371,100 +100,231 @@ class VecSpec extends ChiselPropSpec with Utils {
...
-    }
-  }
-
-  property("A Vec of lit should operate correctly") {
-    forAll(safeUIntN(8)) {
-      case (w: Int, v: List[Int]) =>
-        assertTesterPasses { new LitTester(w, v) }
-    }
-  }
-
```
Comment (jackkoenig) [inline src/test/scala/chiselTests/Vec.scala:387]: This test actually didn't do anything, it passed values but they were ignored and it just used zero. I figure Vecs of Lits are better tested by the Vec literal stuff anyway.

File: src/test/scala/chiselTests/Vec.scala [hunk 2]
```diff
@@ -371,100 +100,231 @@ class VecSpec extends ChiselPropSpec with Utils {
...
 
   property("VecInit should support 3D tabulate bidirectional wire connection") {
-    forAll(smallPosInts, smallPosInts, smallPosInts) { (n: Int, m: Int, p: Int) =>
-      assertTesterPasses { new VecMultiDimTester.BidirectionalTester3DTabulate(n, m, p) }
-    }
-  }
-
-  property("VecInit should iterate correctly") {
-    forAll(Gen.choose(1, 10), smallPosInts) { (start: Int, len: Int) =>
-      assertTesterPasses { new IterateTester(start, len)(x => x + 50.U) }
```
Comment (jackkoenig) [inline src/test/scala/chiselTests/Vec.scala:446]: This was moved to integration tests

File: src/test/scala/chiselTests/Vec.scala [hunk 3]
```diff
@@ -371,100 +100,231 @@ class VecSpec extends ChiselPropSpec with Utils {
...
+          chirrtl should include(s"mods$suffix.io.in <= vec2D[$i][$j][$k].in")
+        }
+      }
     }
   }
 
-  property("Regs of vecs should be usable as shift registers") {
-    forAll(smallPosInts) { (n: Int) => assertTesterPasses { new ShiftRegisterTester(n) } }
-  }
-
```
Comment (jackkoenig) [inline src/test/scala/chiselTests/Vec.scala:453]: Moved to integration tests

File: src/test/scala/chiselTests/Vec.scala [hunk 4]
```diff
@@ -3,79 +3,14 @@
 package chiselTests
 
 import circt.stage.ChiselStage
+import circt.stage.ChiselStage.emitCHIRRTL
 import chisel3._
 import chisel3.testers.{BasicTester, TesterDriver}
 import chisel3.util._
 import org.scalacheck._
```
Comment (seldridge) [inline src/test/scala/chiselTests/Vec.scala:10]: Do these changes remove the dependency on scalacheck in this test or globally?

File: src/test/scala/chiselTests/Vec.scala [hunk 5]
```diff
@@ -371,100 +100,231 @@ class VecSpec extends ChiselPropSpec with Utils {
...
+    chirrtl should include("reg oneBitUnitRegVec : UInt<1>[1], clock")
+    chirrtl should include("oneBitUnitRegVec[0] <= UInt<1>(\"h1\")")
   }
 
   property("A Vec with zero entries should compile and have zero width") {
-    assertTesterPasses { new ZeroEntryVecTester }
-  }
 
-  property("Dynamic indexing of a Vec of Module IOs should work") {
-    assertTesterPasses { new ModuleIODynamicIndexTester(4) }
```
Comment (jackkoenig) [inline src/test/scala/chiselTests/Vec.scala:467]: Also moved to integration tests

**Review by seldridge — APPROVED** [review]
Awesome savings on test runtime.

I glanced through this and the changes seem fine.

File: src/test/scala/chiselTests/Vec.scala (hunk 4 above)
Comment (jackkoenig) [reply src/test/scala/chiselTests/Vec.scala:10]: Just in this test, but it's still a step toward getting rid of it!

**Pull request merged**